components/forms/CreateTxForm/index.tsx

```diff
@@ -2,47 +2,58 @@ import { Account, calculateFee } from "@cosmjs/stargate";
 import { assert } from "@cosmjs/utils";
 import axios from "axios";
 import { NextRouter, withRouter } from "next/router";
-import { useState } from "react";
+import { useRef, useState } from "react";
 import { useAppContext } from "../../../context/AppContext";
-import { capitalizeFirstLetter } from "../../../lib/displayHelpers";
 import { TxMsg, TxType } from "../../../types/txMsg";
 import Button from "../../inputs/Button";
 import Input from "../../inputs/Input";
 import StackableContainer from "../../layout/StackableContainer";
 import MsgForm from "./MsgForm";
 
+export interface MsgGetter {
+  readonly isMsgValid: (msg: TxMsg) => msg is TxMsg;
+  readonly msg: TxMsg;
+}
+
+const getMsgFormKey = (txType: TxType, msg: TxMsg) => JSON.stringify({ txType, msg });
+
 interface CreateTxFormProps {
   readonly router: NextRouter;
-  readonly txType: TxType;
   readonly senderAddress: string;
   readonly accountOnChain: Account;
-  readonly closeForm: () => void;
 }
 
-const CreateTxForm = ({
-  router,
-  txType,
-  senderAddress,
-  accountOnChain,
-  closeForm,
-}: CreateTxFormProps) => {
+const CreateTxForm = ({ router, senderAddress, accountOnChain }: CreateTxFormProps) => {
   const { state } = useAppContext();
 
   const [processing, setProcessing] = useState(false);
-  const [checkAndGetMsg, setCheckAndGetMsg] = useState<() => TxMsg | null>();
+  const [msgTypes, setMsgTypes] = useState<readonly TxType[]>([]);
+  const msgGetters = useRef<MsgGetter[]>([]);
   const [memo, setMemo] = useState("");
   const [gasLimit, setGasLimit] = useState(200000);
   const [gasLimitError, setGasLimitError] = useState("");
 
   const gasPrice = state.chain.gasPrice;
   assert(gasPrice, "gasPrice missing");
 
+  const addTxType = (newTxType: TxType) => {
+    setMsgTypes((oldMsgTypes) => {
+      const newTxTypes = [...oldMsgTypes, newTxType];
+      setGasLimit((oldGasLimit) => (oldGasLimit / (oldMsgTypes.length || 1)) * newTxTypes.length);
+      return newTxTypes;
+    });
+  };
+
   const createTx = async () => {
     try {
       assert(typeof accountOnChain.accountNumber === "number", "accountNumber missing");
-      assert(!!checkAndGetMsg, "form filled incorrectly");
-      const msg = checkAndGetMsg();
-      if (!msg) return;
+      assert(!!msgGetters.current.length, "form filled incorrectly");
```

Comment:
```suggestion
      assert(msgGetters.current.length, "form filled incorrectly");
```

assert checks for truthiness automatically